### DISCUSSION TITLE
How to use VaToast in interceptor

### DISCUSSION THREAD

**Discussion by JohnniePeixoto** [epicmaxco/vuestic-ui#993] [Q&A]
Hello guys, 

I need some help here, i'm trying to call VaToast inside Axios Interceptor, to handle globally some errors.. 
But how do I import VaToast or how to call it from inside the interceptor? 

something like:

```
instance.interceptors.response.use((response) => {
    req.done();
    return Promise.resolve(response);
}, (error) => {
    req.done();
    if (error.response) {
        this.VaToast.init({message: error.response.message, color: 'warning'}) //Here is my problem
    }
    return Promise.reject(error);
});
```

**Comment by m0ksem** [comment]
For now, you can use something like this. That would be great, if we export vaToast from `vuestic-ui`, but it's all we have for now.

```ts
import { getCurrentInstance } from 'vue'

// ...

getCurrentInstance().appContext.config.globalProperties.$vaToast.init({message: error.response.message, color: 'warning'})
```

**Reply by byte-voyager** [reply]
> For now, you can use something like this. That would be great, if we export vaToast from `vuestic-ui`, but it's all we have for now.
> 
> ```ts
> import { getCurrentInstance } from 'vue'
> 
> // ...
> 
> getCurrentInstance().appContext.config.globalProperties.$vaToast.init({message: error.response.message, color: 'warning'})
> ```

base_request.js
```javascript
const service = axios.create({
    baseURL: API_HOST,
    timeout: 60000 * 5
})

service.interceptors.response.use(
(response) => {
        console.log(getCurrentInstance(), '<-getCurrentInstance')  // null
        return response
    }
)
```
